iotdb-core/datanode/src/main/java/org/apache/iotdb/db/utils/datastructure/MergeSortTvListIterator.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.utils.datastructure;
+
+import org.apache.iotdb.db.utils.MathUtils;
+
+import org.apache.tsfile.enums.TSDataType;
+import org.apache.tsfile.file.metadata.enums.TSEncoding;
+import org.apache.tsfile.read.TimeValuePair;
+import org.apache.tsfile.read.reader.IPointReader;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MergeSortTvListIterator implements IPointReader {
+  private final List<TVList.TVListIterator> tvListIterators;
+  private final TSDataType tsDataType;
+  private TSEncoding encoding;
+  private int floatPrecision = -1;
+
+  private TimeValuePair currentTvPair;
+
+  public MergeSortTvListIterator(TSDataType tsDataType, List<TVList> tvLists) {
+    this.tsDataType = tsDataType;
+    tvListIterators = new ArrayList<>();
+    for (TVList tvList : tvLists) {
+      tvListIterators.add(tvList.iterator());
+    }
+  }
+
+  public MergeSortTvListIterator(
+      TSDataType tsDataType, TSEncoding encoding, int floatPrecision, List<TVList> tvLists) {
+    this(tsDataType, tvLists);
+    this.encoding = encoding;
+    this.floatPrecision = floatPrecision;
+  }
+
+  private int getSelectedTVListIndex() {
+    long time = Long.MAX_VALUE;
+    int selectedTVListIndex = -1;
+    for (int i = 0; i < tvListIterators.size(); i++) {
+      TVList.TVListIterator iterator = tvListIterators.get(i);
+      TimeValuePair currTvPair = null;
+      if (iterator.hasNext()) {
+        currTvPair = iterator.current();
+      }
+
+      // update minimum time and remember selected TVList
+      if (currTvPair != null && currTvPair.getTimestamp() <= time) {
+        time = currTvPair.getTimestamp();
+        selectedTVListIndex = i;
+      }
+    }
```

Comment:
Maybe we can not call iter.next in `getSelectedTVListIndex`. For example, we have two tvlist in the following.
```
immuable tvlist -->    (5) 6
working tvlist -->    (1) 2
```
It is supposed to get value 1 for first next. Then index of immutable tvlist remains:
```
immuable tvlist -->    (5) 6
working tvlist -->    1 (2)
```

If we call iter.next, it would be:
```
immuable tvlist -->    5 (6)
working tvlist -->    1 (2)
```